### DISCUSSION TITLE
Chem.MolFromPDBFIle creates broken/disconnected files

### DISCUSSION THREAD

**Discussion by kathputu** [rdkit/rdkit#7905] [Q&A]
Hii Everyone,

I'm trying to use the function Chem.MolFromPDBFile from rdkit to read in chemical structures, however, what I end up seeing is disconnected chemical structure. 
![image](https://github.com/user-attachments/assets/14498593-7960-4f51-87a5-6f94500ce096)

If I take the same .pdb file and open it in pymol, everything looks normal. 
<img width="366" alt="Screenshot 2024-10-08 at 11 18 09 AM" src="https://github.com/user-attachments/assets/770ffff2-eabf-4af8-bf88-385d191e8d0f">

Does anyone know what could be the cause of this and if there's any way to fix this issue? Thank you!!!

Below is the pdb file I used:

HETATM 3532  CAO DRX A 500      62.531  59.770  -3.460  1.00 30.68           C  
HETATM 3533  CAP DRX A 500      61.956  58.869  -2.539  1.00 29.04           C  
HETATM 3534  CAQ DRX A 500      61.316  59.363  -1.401  1.00 27.40           C  
HETATM 3535 CLR  DRX A 500      60.534  58.295  -0.314  1.00 23.58          Cl  
HETATM 3536  CBB DRX A 500      61.265  60.721  -1.129  1.00 31.64           C  
HETATM 3537  CBA DRX A 500      61.870  61.640  -1.970  1.00 30.81           C  
HETATM 3538  CAZ DRX A 500      62.488  61.166  -3.131  1.00 31.39           C  
HETATM 3539  SBC DRX A 500      63.251  62.162  -4.344  1.00 33.07           S  
HETATM 3540  CAY DRX A 500      63.567  60.740  -5.285  1.00 30.11           C  
HETATM 3541  CAN DRX A 500      63.147  59.536  -4.741  1.00 30.86           C  
HETATM 3542  CAM DRX A 500      63.296  58.155  -5.407  1.00 31.94           C  
HETATM 3543  PAD DRX A 500      64.955  57.257  -5.076  1.00 32.41           P  
HETATM 3544  OAB DRX A 500      66.011  58.359  -5.290  1.00 34.17           O1-
HETATM 3545  OAE DRX A 500      65.000  56.114  -6.106  1.00 34.01           O  
HETATM 3546  OAC DRX A 500      64.780  56.771  -3.627  1.00 34.79           O1-
HETATM 3547  CAL DRX A 500      62.900  58.207  -6.894  1.00 31.53           C  
HETATM 3548  OAX DRX A 500      63.632  58.796  -7.690  1.00 31.99           O  
HETATM 3549  NAK DRX A 500      61.762  57.562  -7.228  1.00 30.20           N  
HETATM 3550  CAW DRX A 500      61.247  57.253  -8.499  1.00 30.01           C  
HETATM 3551  CAV DRX A 500      60.038  56.559  -8.492  1.00 28.96           C  
HETATM 3552  CAU DRX A 500      59.493  56.042  -9.684  1.00 29.24           C  
HETATM 3553  CAT DRX A 500      58.308  55.273  -9.667  1.00 27.46           C  
HETATM 3554  CAS DRX A 500      57.838  54.691 -10.836  1.00 28.38           C  
HETATM 3555  CAF DRX A 500      58.504  54.890 -12.037  1.00 28.03           C  
HETATM 3556  CAG DRX A 500      59.648  55.682 -12.100  1.00 27.16           C  
HETATM 3557  CAH DRX A 500      60.171  56.263 -10.918  1.00 28.19           C  
HETATM 3558  CAI DRX A 500      61.359  57.032 -10.918  1.00 27.97           C  
HETATM 3559  CAJ DRX A 500      61.888  57.508  -9.720  1.00 28.90           C  
HETATM 3560 HCAP DRX A 500      61.968  57.806  -2.725  1.00 29.04           H  
HETATM 3561 HCBB DRX A 500      60.749  61.074  -0.248  1.00 31.64           H  
HETATM 3562 HCBA DRX A 500      61.826  62.696  -1.746  1.00 30.81           H  
HETATM 3563 HCAY DRX A 500      64.054  60.864  -6.240  1.00 30.11           H  
HETATM 3564 HCAM DRX A 500      62.567  57.497  -4.937  1.00 31.94           H  
HETATM 3565 HNAK DRX A 500      61.228  57.181  -6.460  1.00 30.20           H  
HETATM 3566 HCAV DRX A 500      59.529  56.340  -7.567  1.00 28.96           H  
HETATM 3567 HCAT DRX A 500      57.781  55.123  -8.737  1.00 27.46           H  
HETATM 3568 HCAS DRX A 500      56.948  54.078 -10.810  1.00 28.38           H  
HETATM 3569 HCAF DRX A 500      58.122  54.405 -12.919  1.00 28.03           H  
HETATM 3570 HCAG DRX A 500      60.155  55.830 -13.042  1.00 27.16           H  
HETATM 3571 HCAI DRX A 500      61.877  57.211 -11.849  1.00 27.97           H  
HETATM 3572 HCAJ DRX A 500      62.824  58.039  -9.773  1.00 28.90           H

**Comment by greglandrum** [comment]
@kathputu that snippet of PDB you included cannot be parsed by the RDKit.
Please provide the PDB file you actually read (you can upload this here by adding `.txt` to the end of the file name).

**Reply by kathputu** [reply]
Hi @greglandrum, thank you for the reply! The pdb file that I'm trying to read is the same content copied to a .txt file like attached. It is the hetatm section sliced from a regular pdb file from rcsb. 

In my experience, most of the ligands are read in fine by rdkit and just those with PO3/PO4 keeps disconnecting. Would you happen to know the reason to this? Thank you once again!

[example.txt](https://github.com/user-attachments/files/17331506/example.txt)

**Reply by greglandrum** [reply]
@kathputu I can't parse that file at all with recent versions of the RDKit.
I get an error about the element Hc not being found, so I think there's a problem with the file.
